cmd/centre/main.go

```diff
@@ -69,35 +70,49 @@ type dserver4 struct {
 	hostFile     string
 }
 
-// lookupIP looks up an IP address corresponding to the given address
-// for mac address, prefix a "u", e.g. "u00:11:22:33:44:55"
-func lookupIP(hostFile string, addr string) ([]net.IP, error) {
+// lookupIP looks up an IP address corresponding to the given name.
+// It also returns a slice of other hostnames it found for the same IP.
+func lookupIP(hostFile string, addr string) (net.IP, []string, error) {
 	var err error
 	// First try the override
 	if hostFile != `` {
 		// Read the file and walk each line looking for a match.
 		// We do this so you can update the file without restarting the server
 		var f *os.File
 		if f, err = os.Open(hostFile); err != nil {
-			return nil, err
+			return nil, nil, err
 		}
+		defer f.Close()
 		// We're going to be real simple-minded. We take each line to consist of
-		// one IP, followed by one or more whitespace-separated hostnames
+		// one IP, followed by one or more whitespace-separated hostnames, with
+		// the mac address at the end:
+		// <ip> <hostname>... <mac>
```

Comment:
as I read this, the mac can be any one of them (which is good) but it means the comment is not right? 
I don't want to require that the mac be last, it's going to lead to confusion if you do that, it would be the only use of hosts I know of where position of a name in the list mattered?

As it is it seems you iterate of the list until a match, then bail, in which case fhs is right, you don't return all the names, which is also potentially confusing for users.